### PR TITLE
Restore gRPC logging

### PR DESCRIPTION
**Pull request by enarha**
The initial implementation was unintentionally removed. Log gRPC authentication enabled/disabled for each call.

### CODE REVIEW

**Comment by enarha** [comment]
Here are two examples of log entries, one for allowed List request and one for denied Delete request:
`{"level":"info","ts":"2023-01-24T12:36:26.227Z","caller":"zap/options.go:212","msg":"finished unary call with code OK","grpc.auth_disabled":false,"grpc.start_time":"2023-01-24T12:36:26Z","system":"grpc","span.kind":"server","grpc.service":"tekton.results.v1alpha2.Results","grpc.method":"ListResults","peer.address":"10.244.1.1:30267","grpc.user":"system:serviceaccount:tekton-pipelines:tekton-results-debug","grpc.issuer":"https://kubernetes.default.svc.cluster.local","grpc.code":"OK","grpc.time_duration_in_ns":3146445}`
`{"level":"info","ts":"2023-01-24T12:36:04.630Z","caller":"zap/options.go:212","msg":"finished unary call with code Unauthenticated","grpc.auth_disabled":false,"grpc.start_time":"2023-01-24T12:36:04Z","system":"grpc","span.kind":"server","grpc.service":"tekton.results.v1alpha2.Results","grpc.method":"DeleteResult","peer.address":"10.244.1.1:37207","grpc.user":"system:serviceaccount:tekton-pipelines:tekton-results-debug","grpc.issuer":"https://kubernetes.default.svc.cluster.local","error":"rpc error: code = Unauthenticated desc = permission denied","grpc.code":"Unauthenticated","grpc.time_duration_in_ns":2246113}`

**Comment by adambkaplan** [comment]
/hold

Per working group call, we want to add an e2e test to ensure we don't regress.

File: cmd/api/main.go [hunk 1]
```diff
@@ -140,10 +148,33 @@ func main() {
 	if err != nil {
 		log.Fatalf("Failed to create server: %v", err)
 	}
+
+	// Shared options for the logger, with a custom gRPC code to log level function.
+	zapOpts := []grpc_zap.Option{
+		grpc_zap.WithDurationField(func(duration time.Duration) zapcore.Field {
+			return zap.Int64("grpc.time_duration_in_ns", duration.Nanoseconds())
```
Comment (adambkaplan) [inline cmd/api/main.go:155]: A bit of a nit - is nanos the right level of precision? Wouldn't milliseconds make more sense for logging (and eventually metrics?)

```suggestion
			return zap.Int64("grpc.time_duration_in_ms", duration.Milliseconds())
```

**Review by adambkaplan — COMMENTED** [review]
/approve

Right path, made one comment about the data that is logged.

**Review by prietyc123 — CHANGES_REQUESTED** [review]

File: test/e2e/e2e_test.go [hunk 2]
```diff
@@ -321,3 +330,73 @@ func recordNames(t *testing.T, records []*resultsv1alpha2.Record) []string {
...
+
+	if strings.Contains(resultsApiLogs, matcher) {
+		t.Logf("Found log match %s\n", matcher)
+	} else {
+		t.Errorf("No match for %s in logs %s", matcher, resultsApiLogs)
+	}
+}
+
+// Returns a list of api pods logs
+func getResultsApiLogs(ctx context.Context, podLogOptions *corev1.PodLogOptions, t *testing.T) (string, error) {
```
[the remark this answers is not in the document]
Comment (enarha) [reply test/e2e/e2e_test.go:364]: @prietyc123 Do you have in mind actual use of that function outside of this test?
My experience with generic `utils` packages is quite bad as they end up as a junkyard for totally unrelated functions which people does not easily find which again leads to code duplication. One advantage I considered in having that function in a separate package is adding a unit test for it. I'll give it a try and see if there is any value as those k8s calls will be mocked anyway. Functional test for it I consider out of scope for this module.

[the remark this answers is not in the document]
Comment (adambkaplan) [reply test/e2e/e2e_test.go:364]: A similar sentiment on calling things "helpers": https://bobcatfish.com/post/helpers/. Let's not prematurely refactor here.

> One advantage I considered in having that function in a separate package is adding a unit test for it.

I'm not particularly keen on unit tests for code that is only used in larger functional/e2e tests. Maybe to ensure that tests fail when they are supposed to?

[the remark this answers is not in the document]
Comment (enarha) [reply test/e2e/e2e_test.go:381]: I added a test which checks for ListResults log entry before the actual gRPC call and fails if any are found. That's in addition to the original test which verifies the log entry exist after the gRPC call.
@prietyc123 @adambkaplan Do you consider that enough and do you have other concerns? If not, please also indicate that on the thread so I know if the thread could be resolved.

File: test/e2e/e2e_test.go [hunk 3]
```diff
@@ -321,3 +330,90 @@ func recordNames(t *testing.T, records []*resultsv1alpha2.Record) []string {
...
+	sinceTime := metav1.Now()
+	podLogOptions := corev1.PodLogOptions{
+		SinceTime: &sinceTime,
+	}
+
+	matcher := "\"grpc.method\":\"ListResults\""
+
+	client := newResultsClient(t, allNamespacesReadAccessPath)
+
+	t.Run("Log entry is not found when not expected", func(t *testing.T) {
```
Comment (prietyc123) [inline test/e2e/e2e_test.go:347]: I think you meant `Log entry is found when not expected`

**Review by prietyc123 — CHANGES_REQUESTED** [review]

File: test/e2e/e2e_test.go (hunk 3 above)
Comment (enarha) [reply test/e2e/e2e_test.go]: Indeed. Corrected.

Comment (prietyc123) [reply test/e2e/e2e_test.go]: LGTM

File: test/e2e/e2e_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (adambkaplan) [reply test/e2e/e2e_test.go:381]: Tests look good to me!

**Comment by adambkaplan** [comment]
@tektoncd/results-collaborators anyone able to take a look at this one?

**Review by khrm — COMMENTED** [review]
/lgtm

I tested this and logs come out fine.

**Review by khrm — COMMENTED** [review]
/kind bug
We need this label for release as discussed in the last WG call.

**Pull request merged**